### PR TITLE
Updated the course & book resources w AI additions

### PR DESCRIPTION
**Pull request by WaunBroderick**
check_urls=free-courses-en.md free-programming-books-en.md

What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

For resources
Free courses and free programming books in the emergent and very popular field of artificial intelligence and graph neural networks.

Why is this valuable (or not)?
I believe this is one of the new blossoming topics in AI and finding resources for this is still difficult though becoming easier as the field exapnds. A lot of these books are still currently pre-published due to how new the knowledge is.

How do we know it's really free?
It is shared in the open commons of McGill and Standford univeristy

For book lists, is it a book? For course lists, is it a course? etc.
It is a book and a course the book is a series of PDFs, and for the course all of the pdf slides and collab docs are open to all.

Checklist:
 Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
 [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
 Include author(s) and platform where appropriate.
 Put lists in alphabetical order, correct spacing.
 Add needed indications (PDF, access notes, under construction).
 Used an informative name for this pull request.

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -198,6 +198,7 @@
 * [IBM AI Engineering Professional Certificate](https://www.coursera.org/professional-certificates/ai-engineer) - Romeo Kienzler, Saeed Aghabozorgi, Joseph Santarcangelo, Alex Aklson et al. (Coursera)
 * [MIT Deep Learning and Artificial Intelligence Lectures](https://deeplearning.mit.edu) - Lex Fridman, et al.
 * [MIT's Artificial Intelligence](http://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-034-artificial-intelligence-fall-2010/) - Prof. Patrick Henry Winston (MIT OpenCourseWare)
+* [Stanford's Machine Learning with Graphs](https://web.stanford.edu/class/cs224w/) - Jurij Leskovac (Stanford University)
```
[the remark this answers is not in the document]
Comment (WaunBroderick) [reply courses/free-courses-en.md]: @eshellman In that case should I submit a new pull request with it removed entirely?

[the remark this answers is not in the document]
Comment (eshellman) [reply courses/free-courses-en.md]: no remove it from this PR

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -134,6 +134,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Artificial Intelligence for a Better Future: An Ecosystem Perspective on the Ethics of AI and Emerging Digital Technologies](https://link.springer.com/book/10.1007/978-3-030-69978-9) - Bernd Carsten Stahl (PDF)
 * [Artificial Intelligence: Foundations of Computational Agents (2010), 1st Edition](https://artint.info/aifca1e.html) - David L. Poole, Alan K. Mackworth @ Cambridge University Press (HTML)
 * [Artificial Intelligence: Foundations of Computational Agents (2017), 2nd Edition](https://artint.info) - David L. Poole, Alan K. Mackworth @ Cambridge University Press (HTML, Slides)
+* [Graph Representational Learning](https://www.cs.mcgill.ca/~wlh/grl_book/) - William L. Hamilton (HTML, PDF)
```
[the remark this answers is not in the document]
Comment (WaunBroderick) [reply books/free-programming-books-subjects.md]: @eshellman changed as per request.

**Comment by eshellman** [comment]
please resolve conflicts.

**Comment by WaunBroderick** [comment]
> 

@eshellman resolved

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**